### DISCUSSION TITLE
next-i18next routing to http://localhost:3000/en despite the defaultLocale being set to es

### DISCUSSION THREAD

**Discussion by is-hiku** [i18next/next-i18next#2068] [Q&A]
Hi everyone,

I have the following `next-i18next.config.js`:

```
const defaultLocale = process.env.DEFAULT_LOCALE || 'es';
module.exports = {
    i18n: {
        defaultLocale: defaultLocale,
        locales: ['es', 'en'],
    },
    fallbackLng: {
        default: ['es']
    },
    localePath: '../../path/to/locales'
}
```

For some reason, when I run the project and visit `http://localhost:3000/` it automatically routes to `http://localhost:3000/en` and the English version of the page is shown.

If I remove `en` from `locales: ['es', 'en'],`. Everything works as expected,`http://localhost:3000/` shows the localized version in Spanish, but then, understandably, the `http://localhost:3000/en` throws a 404.

Any ideas on why this might be happening?

**Comment by is-hiku** [comment]
This **ONLY** happens on the homepage. The other pages appear to be working fine. For example `localhost:3000/categories/somecategory` correctly loads the localized page in Spanish but when I navigate to `/` I am presented with the English version.

Code at `../pages/index.tsx`:
```
export const getStaticProps: GetStaticProps = async ({ locale }) => {
  ...
  return {
    props: {
      ...(await serverSideTranslations(locale ?? 'es', [
        'common',
        'navbar',
      ])),
      blog,
      ...
    },
    revalidate: 10,
  };
};
```

**Reply by adrai** [reply]
What's the value of process.env.DEFAULT_LOCALE ?

**Reply by is-hiku** [reply]
process.env.DEFAULT_LOCALE="es"

**Reply by adrai** [reply]
can you provide a reproducible example repo?
